go/launchd/launchd.go

```diff
@@ -252,21 +259,77 @@ func (s Service) Install(p Plist, wait time.Duration) error {
 	return s.install(p, wait)
 }
 
-func (s Service) savePlist(p Plist) error {
-	plistDest := s.plistDestination()
-
-	if _, ferr := os.Stat(p.binPath); os.IsNotExist(ferr) {
-		return fmt.Errorf("%s doesn't exist", p.binPath)
+func (s Service) checkPlistPaths(p Plist) error {
+	if err := libkb.CanExec(p.binPath); err != nil {
+		s.log.Info("cannot exec binPath %s: %s", p.binPath, err)
+		return err
 	}
 
-	plist := p.plistXML()
+	if p.logPath != "" {
+		// make sure the log directory is writable
+		logDir := filepath.Dir(p.logPath)
+		fi, err := os.Stat(logDir)
+		if err != nil {
+			s.log.Info("log directory %q stat error: %s", logDir, err)
+			return err
+		}
+		if !fi.IsDir() {
+			s.log.Info("log directory %q is not a directory", logDir)
+			return fmt.Errorf("plist logPath error: not a directory %q (full logPath = %q)", logDir, p.logPath)
+		}
+
+		if !writable(logDir) {
+			s.log.Info("log directory %q is not writable by current user", logDir)
+			return fmt.Errorf("log directory %q is not writable by current user (full logPath = %q)", logDir, p.logPath)
+		}
+
+		// log directory looks ok
+		s.log.Info("log directory %q is writable by current user", logDir)
+
+		// make sure the log file is writable if it exists
+		_, err = os.Stat(p.logPath)
+		if err == nil {
+			if !writable(p.logPath) {
+				s.log.Info("log path %q exists but isn't writable by current user", p.logPath)
+				return fmt.Errorf("log path %q exists but isn't writable by current user", p.logPath)
+			}
+			s.log.Info("log path %q exists and is writable by current user", p.logPath)
+		} else if os.IsNotExist(err) {
+			s.log.Info("log path file %q doesn't exist yet (should be ok)", p.logPath)
+		} else {
+			s.log.Info("unexpected stat error on %q: %s", p.logPath, err)
+			return err
+		}
+	}
 
 	// Plist directory (~/Library/LaunchAgents/) might not exist on clean OS installs
 	// See GH issue: https://github.com/keybase/client/pull/1399#issuecomment-164810645
+	plistDest := s.plistDestination()
 	if err := libkb.MakeParentDirs(s.log, plistDest); err != nil {
+		s.log.Info("error making parent directories for %s: %s", plistDest, err)
 		return err
 	}
 
+	plistDir := filepath.Dir(plistDest)
+	if !writable(plistDir) {
+		s.log.Info("plist destination %q is not writable by current user", plistDir)
+		return fmt.Errorf("plist destination dir %q is not writable by current user (full filename = %q)", plistDir, plistDest)
+	}
```

Comment:
There's also the issue that the permissions in launchAgents are too wide open. Maybe let's check for that too? I saw that mentioned on a stack overflow...

Reply:
What do you want to do in that case?

Reply:
at the very least log it, but maybe also fix it as in the other cases?